aeppl/transforms.py

```diff
@@ -135,6 +158,63 @@ def default_transform(fgraph: FunctionGraph, node: Node) -> Optional[List[Node]]
     return trans_node.outputs
 
 
+class TransformValuesMapping(Feature):
+    r"""A `Feature` that maintains a map between value variables and their transforms."""
+
+    def __init__(self, values_to_transforms):
+        self.values_to_transforms = values_to_transforms
+
+    def on_attach(self, fgraph):
+        if hasattr(fgraph, "values_to_transforms"):
+            raise AlreadyThere()
+
+        fgraph.values_to_transforms = self.values_to_transforms
+
+
+class TransformValuesOpt(GlobalOptimizer):
+    r"""Transforms value variables according to a map and/or per-`RandomVariable` defaults."""
+
+    default_transform_opt = in2out(default_transform, ignore_newtrees=True)
+
+    def __init__(
+        self,
+        values_to_transforms: Dict[
+            TensorVariable, Union[RVTransform, DefaultTransformSentinel, None]
+        ],
+    ):
+        """
+        Parameters
+        ==========
+        values_to_transforms
+            Mapping between value variables and their transformations.
+            Each value variable can be assigned one of `RVTransform`,
+            ``DEFAULT_TRANSFORM``, or ``None``. If a transform is not specified for
+            a specific value variable it will not be transformed.
+
+        Raises
+        ======
+        RuntimeError
+            If any of the specified value variables is not successfully transformed
+
+        """
+
+        self.values_to_transforms = values_to_transforms.copy()
```

Comment:
I was copying the dict when passing it to the feature so that the optimizer could be reused without having to reinitialize it (otherwise the class copy of the dict is spent inplace).

I feel that is less error prone (eg if in an iterative environment like Jupyter one makes the opt in a separate cell and runs the joint logprob twice the second one would silently not do anything

Reply:
A test should be added if we want to support this functionality

Reply:
The only logical reason for this step was the check being performed in the original code and its accompanying error, and, since the latter was already questionable (and now just a warning), the `list.pop` and `copy` steps should be removed entirely.